lib/web_ui/lib/src/engine/text_editing/composition_aware_mixin.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html' as html;
+
+import 'text_editing.dart';
+
+
+/// Provides default functionality for listening to HTML composition events.
+///
+/// A class with this mixin generally calls [determineCompositionState] in order to update
+/// an [EditingState] with new composition values; namely, [EditingState.composingBaseOffset]
+/// and [EditingState.composingExtentOffset].
+///
+/// A class with this mixin should call [addCompositionEventHandlers] on initalization, and
+/// [removeCompositionEventHandlers] on deinitalization.
+///
+/// see also:
+///
+/// * [EditingState], the state of a text field that [CompositionAwareMixin] updates.
+/// * [DefaultTextEditingStrategy], the primary implementer of [CompositionAwareMixin].
+mixin CompositionAwareMixin {
+  /// the name of the HTML composition event type that triggers on starting a composition.
+  static const String kCompositionStart = 'compositionstart';
```

Comment:
Could these constants be private?  Or it looks like they're used in tests, so maybe they should be `@visibleForTesting` or even just duplicated in the tests.

Reply:
made private & duplicated in tests. The name should never change anyway since it's defined in the HTML spec